### PR TITLE
[Feature]: Add E2E auth testing, refactor the UI of the Main Screen and Appointments Screen.

### PR DESCRIPTION
**Pull request by Renedz21**
This pull request introduces a comprehensive suite of end-to-end authentication tests using Maestro and refactors the main auth landing screen implementation for improved maintainability and testability. The changes include the addition of multiple YAML test flows covering all major authentication scenarios, a Maestro configuration file, and updates to the `app/(auth)/index.tsx` file to better support testing and modularity.

**Key changes include:**

### 1. Addition of Automated Authentication Test Suite

- Added a master Maestro test suite `.maestro/auth-suite.yaml` that orchestrates the execution of all authentication-related test flows for the app.
- Introduced individual test flows under `.maestro/auth/` covering:
  - Auth landing screen UI and navigation
  - Sign-in and sign-up flows, including form validation and loading states.
  - Social authentication buttons and their interactions
  - Complete user authentication journey
  - Keyboard interaction, accessibility, edge cases, and performance.

### 2. Maestro Test Infrastructure

- Added `.maestro/config.yaml` to configure Maestro test execution, including app ID, retry policy, output format, and screenshot settings.

### 3. Refactoring of Auth Landing Screen

- Refactored `app/(auth)/index.tsx` to import `AuthSheetContent` from a shared component, improving code modularity and maintainability.
- Updated the background to use a gradient view with a TODO note for future image replacement, and improved styling for better visual clarity.
- Added `testID="start_on_app"` to the main CTA button to facilitate reliable test automation.
- Adjusted bottom sheet open animation duration for a snappier user experience.

These changes collectively establish a robust foundation for automated authentication testing and enhance the maintainability and testability of the authentication UI.

### CODE REVIEW

File: app/(tabs)/index.tsx [hunk 1]
```diff
@@ -1,11 +1,171 @@
...
+          <View className="flex-row items-center gap-3">
+            <TouchableOpacity className="p-2 bg-slate-100 rounded-full">
+              <Ionicons
+                name="notifications-outline"
+                size={20}
+                color="#64748B"
+              />
+            </TouchableOpacity>
+            <Image
+              source={{ uri: 'https://i.pravatar.cc/150?img=12' }}
```
[the remark this answers is not in the document]
Comment (Renedz21) [reply app/(tabs)/index.tsx:36]: In this case is only for development

File: app/(tabs)/_layout.tsx [hunk 2]
```diff
@@ -24,6 +24,22 @@ export default function TabLayout() {
...
+          )
+        }}
+      />
+      <Tabs.Screen
+        name="appointments/[appointmentId]/index"
+        options={{
+          title: 'Cita',
+          href: null
+        }}
+      />
```
[the remark this answers is not in the document]
Comment (Renedz21) [reply app/(tabs)/_layout.tsx:42]: I don't get it.

File: app/_layout.tsx [hunk 3]
```diff
@@ -15,7 +15,7 @@ import '../global.css';
 export { ErrorBoundary };
 
 function InitialLayout() {
-  const isAuthenticated = false;
+  const isAuthenticated = true;
```
[the remark this answers is not in the document]
Comment (Renedz21) [reply app/_layout.tsx:18]: In this case is only for development.

File: app/(tabs)/appointments/[appointmentId]/index.tsx [hunk 4]
```diff
@@ -0,0 +1,152 @@
+import { View, Text, TouchableOpacity, ScrollView } from 'react-native';
+
+import { useLocalSearchParams } from 'expo-router';
+
+export default function AppointmentDetails() {
+  const { appointmentId } = useLocalSearchParams();
+  console.log(appointmentId);
```
[the remark this answers is not in the document]
Comment (Renedz21) [reply app/(tabs)/appointments/[appointmentId]/index.tsx:7]: You're right
